src/urlMatcherFactory.js

```diff
@@ -342,6 +342,8 @@ UrlMatcher.prototype.format = function (values) {
 
     if (isPathParam) {
       var nextSegment = segments[i + 1];
+      var isFinalPathParam = i + 1 >= nPath;
```

Comment:
this should probably be changed to `i + 1 === nPath` (`isPathParam` already ensures that this is within `nPath`).